src/neptune/cli/sync.py

```diff
@@ -16,258 +16,103 @@
-        async_path = base_path / ASYNC_DIRECTORY
-        for container_type, unique_id, path in iterate_containers(async_path):
-            if not is_container_synced_and_remove_junk(path):
-                container = get_metadata_container(
-                    backend=self._backend,
-                    container_id=unique_id,
-                    container_type=container_type,
-                )
-                if container:
-                    self.sync_container(container_path=path, experiment=container)
+        if containers.unsynced_containers:
+            for container in containers.unsynced_containers:
+                container.sync(base_path=base_path, backend=backend, project=None)
 
-    def sync_selected_registered_containers(
-        self, base_path: Path, qualified_container_names: Sequence[QualifiedName]
-    ) -> None:
-        for name in qualified_container_names:
-            container = get_metadata_container(
-                backend=self._backend,
-                container_id=name,
+        if containers.offline_containers:
+            project = get_project(
+                project_name_flag=QualifiedName(project_name) if project_name else None, backend=backend
             )
-            if container:
-                container_path = base_path / ASYNC_DIRECTORY / f"{container.type.create_dir_name(container.id)}"
-                container_path_deprecated = base_path / ASYNC_DIRECTORY / f"{container.id}"
-                if container_path.exists():
-                    self.sync_container(container_path=container_path, experiment=container)
-                elif container_path_deprecated.exists():
-                    self.sync_container(container_path=container_path_deprecated, experiment=container)
-                else:
-                    logger.warning("Warning: Run '%s' does not exist in location %s", name, base_path)
+            if not project:
+                raise CannotSynchronizeOfflineRunsWithoutProject
 
-    def _register_offline_container(self, project: Project, container_type: ContainerType) -> Optional[ApiExperiment]:
-        try:
-            if container_type == ContainerType.RUN:
-                return self._backend.create_run(project.id)
-            else:
-                raise ValueError("Only runs are supported in offline mode")
-        except Exception as e:
-            logger.warning(
-                "Exception occurred while trying to create a run" " on the Neptune server. Please try again later",
-            )
-            logging.exception(e)
-            return None
+            for container in containers.offline_containers:
+                container.sync(base_path=base_path, backend=backend, project=project)
 
     @staticmethod
-    def _move_offline_container(
-        base_path: Path,
-        offline_dir: str,
-        server_id: UniqueId,
-        server_type: ContainerType,
+    def sync_selected(
```

Comment:
As far as I understand, this method does two things:
- syncs async containers
- syncs offline containers

I wonder if we could create two separate functions to handle those separately and then invoke them in this method? It would be easier to read. What do you think?

Reply:
Sure. Done :wink: